httpx/middleware/chain.go

```diff
@@ -0,0 +1,56 @@
+package middleware
+
+import (
+	"net/http"
+
+	"github.com/remind101/pkg/httpx"
+	"golang.org/x/net/context"
+)
+
+// Inspiration from https://github.com/justinas/alice
+
+type Link func(httpx.Handler) httpx.Handler
+
+type Chain struct {
+	links []Link
+}
+
+// NewChain creates a chain of "Link"s.
+// Each Link represents a middleware
+func NewChain(links ...Link) Chain {
```

Comment:
Now that I've had more time to look at this, I'm not sure this actually adds that much value. Taking the example in alice's readme, these are equivalent:

```go
// alice
h := alice.New(th.Throttle, timeoutHandler, nosurfe.NewPure).Then(myHander)

// standard
h := middleware(myHandler)

func middleware(h http.Handler) http.Handler {
        // Add throttling
        h = throttled.Interval(throttled.PerSec(10), 1, &throttled.VaryBy{Path: true}, 50).Throttle(h)

        // Timeout requests
        h = timeoutHandler(h)

        return nosurf.New(h)
}
```

The difference is that with alice, all of your middleware needs to be "alice aware" and provide either a `func(http.Handler) http.Handler`, or a `func(http.Handler) http.Handler` factory method (e.g. [nosurf.NewPure](https://github.com/justinas/nosurf/blob/master/handler.go#L105-L108)). In the end, it kinda feels like you end up having to write more code than just having a function that encapsulates the middleware to wrap a handler with.